src/argilla/client/workspaces.py

```diff
@@ -0,0 +1,64 @@
+#  coding=utf-8
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import TYPE_CHECKING, Optional
+
+import argilla as rg
+
+if TYPE_CHECKING:
+    from argilla.client.api import Argilla
+
+
+class Workspace:
+    """The `Workspace` class is used to manage workspaces in Argilla. The main
+    purpose of this class is to ease the conversion from name to ID, since the
+    workspace name is known to the user while the ID is not. Also, it helps checking
+    whether the workspace exists or not.
+
+    Args:
+        name: the name of the workspace to be managed. If not provided, the current
+            workspace will be used.
+
+    Raises:
+        ValueError: if the workspace does not exist in the current Argilla account.
+
+    Examples:
+        >>> from argilla import rg
+        >>> workspace = rg.Workspace("my-workspace")
+        >>> workspace.id
+    """
+
+    def __init__(self, name: Optional[str] = None) -> None:
```

Comment:
As for the dataset initialization, I would prefer to remove this code from the __init__ class method. For me, make sense having a  class method to fetch the workspace by name:

```python
workspace = rg.Workspace.find_by_name(name="...")
...
```

Later, we can user the __init__ method to create an workspace instance and use it create a new workspace in argilla:

```python
new_workspace = rg.Workspace(name="new-workspace")

new_workspace.save()
# or:
new_workspace.create()
# or:
rg.Workspace.save(new_workspace)
# or:
rg.create_workspace(new_workspace)
# or (sorry for the alternatives,...)
from argilla import workspaces

workspaces.create(new_workspace)
```

Reply:
Ok it makes sense if we intended to extend it! Thanks for the feedback 👍🏻 Do you agree on naming the classmethod `rg.Workspace.from_name` instead of `rg.Workspace.find_by_name` since the first is probably the most common used in Python. Unless you have any special preference for `find_by_name` obviously.